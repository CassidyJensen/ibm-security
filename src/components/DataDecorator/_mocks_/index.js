/**
 * @file Data decorator mocks.
 * @copyright IBM Security 2019
 */

export default {
  type: 'IP',
  value: '10.0.0.0',
  score: 0,
  href: '#',
};
