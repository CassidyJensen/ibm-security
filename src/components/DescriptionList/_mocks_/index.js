/**
 * @file Description list mocks.
 * @copyright IBM Security 2019 - 2020
 */

export default [
  {
    id: 0,
    title: 'Name',
    description: 'Some text entered here',
  },
  {
    id: 1,
    title: 'Longer Name',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc dui magna, finibus id tortor sed, aliquet bibendum augue. Aenean posuere sem vel euismod dignissim. Nulla ut cursus dolor. Pellentesque vulputate nisl a porttitor interdum. ',
  },
  {
    id: 2,
    title: 'Even Longer Name',
    description:
      'Sed ut et perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam.',
  },
];
