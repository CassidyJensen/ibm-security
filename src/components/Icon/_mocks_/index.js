/**
 * @file Icon mocks.
 * @copyright IBM Security 2019
 */

import Add16 from '@carbon/icons-react/lib/add/16';

const path = 'M0';
const renderIcon = Add16;

export { path, renderIcon };
