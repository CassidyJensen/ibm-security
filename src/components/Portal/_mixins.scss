////
/// Portal mixins.
/// @group portal
/// @copyright IBM Security 2019
////

@import '../../globals/index';

@mixin portal {
  &__overlay {
    display: flex;
    position: fixed;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    background-color: $overlay-01;
    justify-content: center;
    z-index: z($layer: overlay);

    &:focus {
      outline: unset;
    }
  }

  &__container {
    overflow-y: hidden;
  }
}
