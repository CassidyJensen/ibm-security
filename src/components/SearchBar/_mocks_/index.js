/**
 * @file Search bar mocks.
 * @copyright IBM Security 2019
 */

const scopes = new Array(3)
  .fill()
  .map((name = 'Scope', id) => ({ id, name: `${name} ${id + 1}` }));

export default {
  submitLabel: 'Submit',
  placeHolderText: 'Placeholder',
  labelText: 'Label',
  clearButtonLabelText: 'Clear',
  value: 'Initial value',
  scopes,
  selectedScopes: [scopes[0]],
  scopesTypeLabel: 'Scopes',
  allSelectedScopesLabel: 'All scopes',
};
