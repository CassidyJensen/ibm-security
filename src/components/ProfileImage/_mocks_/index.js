/**
 * @file Profile image mocks.
 * @copyright IBM Security 2019
 */

import { className } from '../../_mocks_';
import { profile } from '../../Header/_mocks_';

const imageUrl = 'http://via.placeholder.com/160x160';

const profileWithImage = Object.assign({}, profile, {
  image_url: imageUrl,
});

const large = true;

export { className, large, profile, profileWithImage };
