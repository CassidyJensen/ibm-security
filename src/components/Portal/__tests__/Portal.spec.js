/**
 * @file Portal tests.
 * @copyright IBM Security 2018
 */

import { mount, shallow } from 'enzyme';
import React from 'react';

import { className, label } from '../../_mocks_';

import Portal from '../';

describe('Portal', () => {
  let portal;

  beforeEach(() => {
    portal = (
      <Portal className={className} title={label}>
        <div>
          <button>TEST</button>
        </div>
      </Portal>
    );
  });

  it('renders correctly', () => {
    expect(mount(portal)).toMatchSnapshot();
  });

  it("doesn't render", () => {
    expect(shallow(portal)).toMatchSnapshot();
  });
});
