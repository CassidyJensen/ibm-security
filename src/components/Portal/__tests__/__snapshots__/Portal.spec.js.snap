// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portal doesn't render 1`] = `
<Portal
  containerInfo={
    <body
      class=""
    >
      <div>
        <button>
          TEST
        </button>
      </div>
      <div
        class="security--portal__overlay"
        tabindex="-1"
      />
      <div
        class="security--portal__overlay"
        tabindex="-1"
      />
    </body>
  }
>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={true}
    focusTrapOptions={
      Object {
        "initialFocus": null,
      }
    }
    paused={false}
  >
    <div>
      <button>
        TEST
      </button>
    </div>
  </FocusTrap>
</Portal>
`;

exports[`Portal renders correctly 1`] = `
<Portal
  className="class"
  focusTrap={true}
  hasOverlay={true}
  initialFocus={null}
  rootNode={
    <body
      class="security--portal__container"
    >
      <div>
        <button>
          TEST
        </button>
      </div>
      <div
        class="security--portal__overlay"
        tabindex="-1"
      />
    </body>
  }
  stopPropagation={false}
  title="Label"
>
  <Portal
    containerInfo={
      <body
        class="security--portal__container"
      >
        <div>
          <button>
            TEST
          </button>
        </div>
        <div
          class="security--portal__overlay"
          tabindex="-1"
        />
      </body>
    }
  >
    <FocusTrap
      _createFocusTrap={[Function]}
      active={true}
      focusTrapOptions={
        Object {
          "initialFocus": null,
        }
      }
      paused={false}
    >
      <div>
        <button>
          TEST
        </button>
      </div>
    </FocusTrap>
  </Portal>
</Portal>
`;
